src/Type/Constant/ConstantArrayTypeBuilder.php

```diff
@@ -61,7 +65,25 @@ public static function createFromConstantArray(ConstantArrayType $startArrayType
 		);
 	}
 
-	public function setOffsetValueType(?Type $offsetType, Type $valueType, bool $optional = false): void
+	private function getNextIntKeyFromConstantType(): int
+	{
+		$count = 0;
+		/** @var ConstantIntegerType|ConstantStringType $keyType */
+		foreach ($this->keyTypes as $keyType) {
+			if (!is_numeric($keyType->getValue())) {
```

Comment:
This is what `$this->nextAutoIndex` is already for

Reply:
I don't think that ConstantArrayTypeBuilder should be touched by this PR at all, all the logic should be in the array_replace extension.

Reply:
@ondrejmirtes done, thanks for the hint. :)